Hw3/IQueue.cs

```diff
@@ -0,0 +1,30 @@
+public interface IQueue<T>
+{
+    public void Enqueue(T t);
+    public T Dequeue();
+    public T Peek();
+    public bool IsFull();
+    public bool IsEmpty();
+    public int GetSize();
+}
+
+public static class QueueExtensions
+{
+    public static IQueue<T> Tail<T>(this IQueue<T> queue)
+    {
+        if (queue.IsEmpty())
+        {
+            throw new InvalidOperationException("Queue is empty.");
+        }
+
+        IQueue<T> newQueue = new MyQueue<T>(queue.GetSize());
+        queue.Dequeue();
+
+        while (!queue.IsEmpty())
+        {
+            newQueue.Enqueue(queue.Dequeue());
+        }
```

Comment:
It seems that original queue is empty now, this is not proper behavior (restore it to an initial state)